src/dms/core/EdFi.DataManagementService.Core/ApiSchema/ApiSchemaFileLoader.cs

```diff
@@ -24,105 +25,116 @@ private static JsonNode LoadFromAssembly(string resourceName, Assembly assembly)
+                {
+                    throw new InvalidOperationException($"Access denied to file '{filePath}'", ex);
+                }
+            }
+        }
+        catch (DirectoryNotFoundException ex)
+        {
+            throw new InvalidOperationException($"Directory not found: '{directoryPath}'", ex);
+        }
+        catch (UnauthorizedAccessException ex)
+        {
+            throw new InvalidOperationException($"Access denied to directory '{directoryPath}'", ex);
         }
-    });
 
-    public JsonNode CoreApiSchemaRootNode => _coreApiSchemaRootNode.Value;
+        return fileContents;
+    }
 
     /// <summary>
-    /// Loads extension ApiSchema files if they exist, either from the file system or ApiSchema assemblies
+    /// Returns core and extension ApiSchema JsonNodes
     /// </summary>
-    private readonly Lazy<JsonNode[]> _extensionApiSchemaRootNodes = new(() =>
+    private readonly Lazy<ApiSchemaNodes> _apiSchemaNodes = new(() =>
     {
-        _logger.LogDebug("Entering ApiSchemaFileLoader._extensionApiSchemaRootNode");
-
-        if (appSettings.Value.UseLocalApiSchemaJson)
+        if (appSettings.Value.UseApiSchemaPath)
         {
-            string? jsonContent;
-            try
-            {
-                jsonContent = File.ReadAllText(
-                    Path.Combine(
-                        Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location) ?? "",
-                        "ApiSchema",
-                        "ApiSchema.Extension.json"
-                    )
-                );
-
-                // No extension file found
-                if (jsonContent == null)
-                {
-                    return [];
-                }
-            }
-            catch (Exception ex)
-            {
-                _logger.LogInformation(ex, "No ApiSchema.Extension.json file was found.");
-                return [];
-            }
+            string apiSchemaPath =
+                appSettings.Value.ApiSchemaPath
+                ?? throw new InvalidOperationException("No ApiSchemaPath configuration is set");
+            List<JsonNode> apiSchemaNodes = ReadApiSchemaFiles(apiSchemaPath);
 
-            JsonNode rootNode =
-                JsonNode.Parse(jsonContent)
-                ?? throw new InvalidOperationException("Unable to parse ApiSchema extension file");
-            return [rootNode];
+            JsonNode coreApiSchemaNode = apiSchemaNodes.First(node =>
+                !node.SelectRequiredNodeFromPathAs<bool>("$.projectSchema.isExtensionProject", _logger)
+            );
+
+            JsonNode[] extensionApiSchemaNodes = apiSchemaNodes
+                .Where(node =>
+                    node.SelectRequiredNodeFromPathAs<bool>("$.projectSchema.isExtensionProject", _logger)
+                )
+                .ToArray();
+
+            return new(coreApiSchemaNode, extensionApiSchemaNodes);
         }
         else
         {
```

Comment:
Do we still need the "UseApiSchemaPath" flag? Aren’t we always loading the files from the configured "ApiSchemaPath"?